### DISCUSSION TITLE
NextJS, remove elements onClick

### DISCUSSION THREAD

**Discussion by smultar** [vercel/next.js#21313] [Help]
I'm trying to have a component remove itself and its children on click. If someone could point me to documentation and provide an example of how to do this, it would greatly help.

I'm fairly new to React and Nextjs. I'm coming from an ExpressJS with a Jquery background.

`import Link from 'next/link';
import React, { Component } from "react";

export default function AccountOption({ scriptID, link, image, type, name, price }) {

    return (
        <cart-item scriptID={scriptID}>
            <Link href={`${link}`}>
                <img className="cart-item-image" src={image}></img>
            </Link>

            <Link href={`${link}`}>
            <div>
                <p className="cart-item-type">{type}</p>
                <p className="cart-item-name">{name}</p>
            </div>
            </Link>

            <p>{price ? price : "n/a" }</p>
            <a className="cart-item-remove" >✕</a>
        </cart-item>
    )
  }`
  
  Really unsure of how to approach this.

**Comment by toyamarinyon** [comment]
I think you should to read this thread first.
https://stackoverflow.com/questions/35994313/how-component-can-remove-delete-itself-from-dom/35994530

The important part is this.
> the container component (ie List) handle the onDelete functionality, and then update its state 

Once you have a general understanding of the relationship between containers and components, you can take a look ad my sample app for reference.

Code: https://github.com/toyamarinyon/nextjs-list-sample
Demo: https://nextjs-list-sample.vercel.app/

In practice, it's a bit more complicated because we need to communicate with the server and manage the state, but the part that deletes the component looks like this.
